pkg/app/app.go

```diff
@@ -1830,6 +1830,24 @@ func (a *App) template(r *Run, c TemplateConfigProvider) (bool, []error) {
 		}
 	}
 
+	// We need to forcefully traverse release dependencies to be able to template single release with dependencies
+	// We don't want to use a full list to prevent Helmfile becomes too slow when there are many releases
+	batches, err = st.PlanReleases(state.PlanOptions{Reverse: false, SelectedReleases: selectedReleases, IncludeNeeds: true, IncludeTransitiveNeeds: true, SkipNeeds: false})
+	if err != nil {
+		return false, []error{err}
+	}
+
+	var traversedReleasesWithNeeds []state.ReleaseSpec
+	for _, rs := range batches {
+		for _, r := range rs {
+			traversedReleasesWithNeeds = append(traversedReleasesWithNeeds, r.ReleaseSpec)
+		}
+	}
+
+	// Traverse DAG of all the releases so that we don't suffer from false-positive missing dependencies
+	// withDAG require dependencies to be present, in case of strict selector we should include them anyway
+	st.Releases = traversedReleasesWithNeeds
```

Comment:
@kuzaxak Hey! Thanks for the PR. The other half of your changes looks good. But I can't see why this one is needed.
Saw your `template_a_single_release_with_dependency` test case- shouldn't you just specify `--skip-needs` if you don't want to include the dep?

Reply:
Hey, we discussed that in your PR [here][1] but this PR closed

[1]: https://github.com/helmfile/helmfile/pull/78

Reply:
@mumoshu I expect yes, but in any case I will test it right after merge. Based on your tests cases it should work as you described.

I will rebase and delete commit with graph changes.

Reply:
@kuzaxak #78 has been merged. Would you mind giving it a shot and remove the graph changes if it looks good?

Reply:
Cool, yes, I will rebase my branch and remove graph related changes.

Reply:
I'm not even sure is it needed anymore due to merged [PR][1]. Will test and close if not needed. 

[1]: https://github.com/helmfile/helmfile/pull/47